Source/WebCore/dom/Observable.cpp

```diff
@@ -169,6 +171,15 @@ void Observable::some(ScriptExecutionContext& context, Ref<PredicateCallback>&&
     return createInternalObserverOperatorSome(context, *this, WTFMove(callback), options, WTFMove(promise));
 }
 
+void Observable::reduce(ScriptExecutionContext& context, Ref<ReducerCallback>&& callback, std::optional<JSC::JSValue>&& initialValue, const SubscribeOptions& options, Ref<DeferredPromise>&& promise)
+{
+    // IDL for 'optional any' defaults to 'jsUndefined',
+    // so std::optional is always present. If undefined, set to nullopt.
+    if (initialValue->isUndefined())
+        initialValue = std::nullopt;
```

Comment:
This felt a little funky; the only way I got an `IDLOptional` to generate in the JS* classes, was with `optional object` than `optional any`.

Does `optional any` suggest that; `reduce(fn)` and `reduce(fn, undefined)` are equivalent, and if so, is checking for "undefined" enough to suffice the spec requirement in [point 7](https://wicg.github.io/observable/#dom-observable-reduce)

Reply:
The question of whether `reduce(fn)` and `reduce(fn, undefined)` are equivalent is best answered with test cases so we know it’s interoperable. For as long as I can remember I’ve seen questions about whether `undefined` and a missing argument behave the same, and the answer has been different in different contexts. For example, things built into the JavaScript language seem to definitely treat them the same, but not sure DOM always does. If we have a test case, then we can worry about how we are coding it and not whether the behavior is correct.

In terms of how we write the code, if the bindings generated code never passes `std::nullopt`, then we should just use `JSC::JSValue`, and not involve `std::optional`. We can always switch to using `std::optional` later if we change how the bindings generate. It’s not helpful to use rvalue references with `std::optional<JSC::JSValue>`, by the way, since it’s all scalars and no helpful meaning to the move semantic, so we should probably not use `&&`.